File: Service/Models/WagerAuditEntryModel.cs
using TecmoTourney;

namespace TecmoTourney.Models
{
    public class WagerAuditEntryModel
    {
        public int AuditId { get; set; }
        public int? TournamentId { get; set; }
        public int TargetPlayerId { get; set; }
        public int? ActorPlayerId { get; set; }
        public WagerAuditAction Action { get; set; }
        public int? WagerId { get; set; }
        public int? GameResultId { get; set; }
        public decimal? Amount { get; set; }
        public decimal? BalanceBefore { get; set; }
        public decimal? BalanceAfter { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: Client/tecmo-wager/src/app/core/models/wager-audit-entry.model.ts
export type WagerAuditAction =
  | 'PlaceWager'
  | 'CancelWager'
  | 'SettleWagerWin'
  | 'SettleWagerLose'
  | 'VoidWager'
  | 'BalanceSet'
  | 'BalanceAdd'
  | 'BalanceSetToZero'
  | 'AdminCancelWager';

export interface WagerAuditEntry {
  auditId: number;
  tournamentId: number | null;
  targetPlayerId: number;
  actorPlayerId: number | null;
  action: string;
  wagerId: number | null;
  gameResultId: number | null;
  amount: number | null;
  balanceBefore: number | null;
  balanceAfter: number | null;
  createdAt: string;
}
